airflow/dag_processing/processor.py

```diff
@@ -365,8 +367,10 @@ def __init__(self, dag_ids: list[str] | None, dag_directory: str, log: logging.L
         self._dag_directory = dag_directory
         self.dag_warnings: set[tuple[str, str]] = set()
 
+    @staticmethod
+    @internal_api_call
     @provide_session
-    def manage_slas(self, dag: DAG, session: Session = None) -> None:
+    def manage_slas(dag_folder, dag_id: str, session: Session = None) -> None:
```

Comment:
```suggestion
    def manage_slas(dag_folder, dag_id: str, session: Session = NEW_SESSION) -> None:
```

Good change to fix this

Reply:
Good catch!